

**1.  Breast Cancer Detection Model(Benign or Malignant)**




#### I will take breast cancer dataset from the sklearn library implement support vector machine algorithm and find the accuracy for my model.

In [ ]:
# Imported libraries 
#Printed the labels
#Imported the dataset
import numpy as np
import matplotlib.pyplot as plt
from sklearn import svm
from sklearn import datasets
cancer = datasets.load_breast_cancer()
print("Labels: ", cancer.target_names)
from sklearn.metrics import confusion_matrix

Labels:  ['malignant' 'benign']


### I have imported libraries, the dataset and printed out the labels


In [ ]:
# Split the dataset using train_test-split from sklearn
# Model evaluation using accuracy score and confusion matrix
# Import SVC and fit the training data using linear kernel
# Make a prediction on y-test
# Calculate the accuracy
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(cancer.data, cancer.target, test_size=0.4,random_state=99)
classifier = svm.SVC(kernel='linear')
 
classifier.fit(X_train, y_train)
y_pred = classifier.predict(X_test)
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
confusion_matrix = confusion_matrix(y_test, y_pred)
confusion_matrix

Accuracy: 0.9605263157894737


array([[ 73,   3],
       [  6, 146]])

.      Splitted the dataset using train_test-split from sklearn.

For evaluation of the model imported accuracy_score and confusion matrix

Initiated object for SVC that is svc_model and fitted the training data to the model. Used ‘linear’ as a kernel.

Made a prediction on y_test and calculated accuracy score on test data 

The accuracy is 96%.
 73 positve class data points were correctly classified by the model

 146 negative class datapoints were correctly classified by the model

 3 negative class data points were incorrectly classified as belonging to the positive class

 6 positive class points were incorrectly classified as belonging to the negative class of the model


**2.   Passengers Survived or Not Model(Titanic Dataset)**



In [ ]:
# Imported libraries 
#Printed the labels
#Imported the dataset
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn import svm
from sklearn import datasets
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import confusion_matrix

In [ ]:
#Load the dataset
# print the labels
titanicc=pd.read_csv("/content/drive/MyDrive/Titanic.csv")
titanicc.head()

print("Labels: ", titanicc.Survived)

Labels:  0      0
1      1
2      0
3      0
4      1
      ..
413    0
414    1
415    0
416    0
417    0
Name: Survived, Length: 418, dtype: int64


In [ ]:
# converting to numerical dummies and assigning to variables
Name = titanicc[['Name']].apply(LabelEncoder().fit_transform)
Sex = titanicc[['Sex']].apply(LabelEncoder().fit_transform)
Cabin = titanicc[['Cabin']].apply(LabelEncoder().fit_transform)
Ticket = titanicc[['Ticket']].apply(LabelEncoder().fit_transform)
Embarked = titanicc[['Embarked']].apply(LabelEncoder().fit_transform)



In [ ]:
# combining variables for numerical dummies to one dataFrame
titanicc_data = pd.concat([Name,Sex, Cabin,Ticket,Embarked], axis = 1)

In [ ]:
titanicc_data.head()

Name  Sex  Cabin  Ticket  Embarked
0   206    1     76     152         1
1   403    0     76     221         2
2   269    1     76      73         1
3   408    1     76     147         2
4   178    0     76     138         2

In [ ]:
titanicc_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   Name      418 non-null    int64
 1   Sex       418 non-null    int64
 2   Cabin     418 non-null    int64
 3   Ticket    418 non-null    int64
 4   Embarked  418 non-null    int64
dtypes: int64(5)
memory usage: 16.5 KB


In [ ]:
#Split and fit the model

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(titanicc_data, titanicc.Survived, test_size=0.4,random_state=99)
classifier = svm.SVC(kernel='linear')
 
classifier.fit(X_train, y_train)
y_pred = classifier.predict(X_test)
# Model evaluation
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
confusion_matrix = metrics.confusion_matrix(y_test,y_pred)

confusion_matrix

Accuracy: 1.0


array([[102,   0],
       [  0,  66]])